portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,42 +14,40 @@
 
 package com.google.sps.servlets;
 
+import static com.googlecode.objectify.ObjectifyService.ofy;
+
 import com.google.sps.data.Comment;
 import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
 import java.io.IOException;
-import java.util.Deque;
-import java.util.ArrayDeque;
+import java.util.List;
 import java.util.Date;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.ws.rs.core.MediaType;
 
-/** Servlet that returns some example content. */
+/** Servlet that returns and saves the comments. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private final Deque<Comment> comments = new ArrayDeque<>();
-
-  public DataServlet() {
-    // adds sample comments
-    comments.addFirst(new Comment("Loris", "Nice!", new Date()));
-    comments.addFirst(new Comment("Mark", "Very interesting.", new Date()));
-  }
-
   /**
-   * Converts a Deque instance into a JSON string.
+   * Converts a List instance into a JSON string.
    */
-  private String convertDequeToJson(Deque<Comment> comments) {
-    Gson gson = new Gson();
+  private String convertListToJson(List<Comment> comments) {
+    Gson gson = new GsonBuilder().setDateFormat("MMM d, yyyy HH:mm z").create();
```

Comment:
extract the date format into a class constant. You might even want to use a constant provided by the standard library so it could correctly factor in the user locale.
Something like https://docs.oracle.com/javase/8/docs/api/java/text/DateFormat.html
```
 DateFormat df = DateFormat.getDateInstance(DateFormat.LONG, Locale.FRANCE);
```
(of for now just omit the optional param for locale if you don't want to build logic to infer this from the request)